app/src/usb_hid.c

```diff
@@ -38,6 +92,27 @@ int zmk_usb_hid_send_report(const uint8_t *report, size_t len) {
         return -ENODEV;
     default:
         k_sem_take(&hid_sem, K_MSEC(30));
+        uint8_t *report;
+        size_t len;
+
+        if (report_id == HID_REPORT_ID_KEYBOARD) {
+#if IS_ENABLED(CONFIG_ZMK_USB_BOOT)
+            report = zmk_hid_get_keyboard_report(HID_REPORT_FULL, hid_protocol);
```

Comment:
Same as above. If the implementations of these functions were changed, this code could easily become wrong without causing any obvious compiler error.